### PR TITLE
ref: error parsing & handling

### PR DESCRIPTION
**Pull request by crisog**
This PR includes a refactor on error parsing & an additional feature that doesn't allow invalid responses (non JSON-RPC) to be relayed to clients/users.

UPDATE: also added some facets when logging 'failed reaching altruist'  eth_logs limit error.

### CODE REVIEW

File: src/controllers/v1.controller.ts [hunk 1]
```diff
@@ -304,8 +304,7 @@ export class V1Controller {
     @param.filter(Applications, { exclude: 'where' })
     filter?: FilterExcludingWhere<Applications>
   ): Promise<string | ErrorObject> {
-    const parsedRawData = parseRawData(rawData)
```
Comment (crisog) [inline src/controllers/v1.controller.ts:307]: Having this outside of the try-catch, made our code trigger Loopback http errors thats why sometimes you'd get non JSON RPC errors (not desired).

File: src/controllers/v1.controller.ts [hunk 2]
```diff
@@ -267,7 +267,7 @@ export class V1Controller {
         origin: this.origin,
```
Comment (nymd) [inline src/controllers/v1.controller.ts:267]: import AWS at line 1 can be deleted

File: src/services/limiter.ts [hunk 3]
```diff
@@ -67,7 +68,10 @@ export async function enforceEVMLimits(
           fromBlock = latestBlock
         }
       } catch (e) {
-        logger.log('error', `Failed trying to reach altruist (${altruistUrl}) to fetch block number.`)
+        logger.log('error', `Failed trying to reach altruist (${altruistUrl}) to fetch block number.`, {
```
Comment (nymd) [inline src/services/limiter.ts:71]: Nice

File: src/services/pocket-relayer.ts [hunk 4]
```diff
@@ -242,13 +240,23 @@ export class PocketRelayer {
           })
 
           if (!(relayResponse instanceof Error)) {
+            const parsedRelayResponse = jsonrpc.parse(relayResponse.payload as string) as IParsedObject
+
+            if (parsedRelayResponse.type === 'invalid' && parsedRelayResponse.payload.message === 'Parse error') {
```
Comment (nymd) [inline src/services/pocket-relayer.ts:249]: This could use some comments describing it

**Review by nymd — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -70,6 +70,7 @@ If you ever get an error from the API, it will be in JSON-RPC format. This is ou
 | -32063 | Please use an explicit block number instead of 'latest' | eth_getLogs Limit Error | non-standard |
 | -32064 | You cannot query logs for more than X amount of blocks | eth_getLogs Limit Error | non-standard |
 | -32065 | Node returned an invalid response | The node serving your request returned an invalid response | non-standard |
+| -32066 | Your request body is not proper JSON | We couldn't parse your request body | non-standard |
```
Comment (rem1niscence) [inline README.md]: as general, I'd rather have a more indirect response type of answer, something more like `The request body is not proper JSON | Request bould couldn't be parsed`. Overall avoid addressing the user directly

File: src/services/pocket-relayer.ts [hunk 6]
```diff
@@ -350,7 +358,8 @@ export class PocketRelayer {
         throw e
       }
 
-      logger.log('error', 'ERROR relaying through node: ' + e, {
+      // Any other error (f.g parsing errors) that should not be propagated as response
```
Comment (rem1niscence) [inline src/services/pocket-relayer.ts]: what f.g mean?

**Review by rem1niscence — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -70,6 +70,7 @@ If you ever get an error from the API, it will be in JSON-RPC format. This is ou
 | -32063 | Please use an explicit block number instead of 'latest' | eth_getLogs Limit Error | non-standard |
 | -32064 | You cannot query logs for more than X amount of blocks | eth_getLogs Limit Error | non-standard |
 | -32065 | Node returned an invalid response | The node serving your request returned an invalid response | non-standard |
+| -32066 | The request body is not proper JSON | Request bould couldn't be parsed | non-standard |
```
Comment (nymd) [inline README.md:73]: typo: request ~~bould~~ body

**Review by nymd — CHANGES_REQUESTED** [review]
one typo

File: README.md (hunk 7 above)
Comment (crisog) [reply README.md:73]: Fixed in develop, merged without noticing the typo

**Pull request merged**